jcl/src/java.base/share/classes/java/lang/StackTraceElement.java

```diff
@@ -57,6 +59,8 @@ public StackTraceElement(String cls, String method, String file, int line) {
 	moduleName = null;
 	moduleVersion = null;
 	classLoaderName = null;
+	includeClassLoaderName = true;
```

Comment:
I have these set to `true` since for publicly constructed `StackTraceElement` objects, this information is to be included in the stack trace. The printing method in Util handles checking for null/empty values.

I could set these to `false` and add another flag eg. `createdFromPublicConstructor` to set those flags to `true` when `toString()` is called on the `StackTraceElement` if it is true. Util will have to be updated to check for `createdFromPublicConstructor` in case the stack trace is printed without using `StackTraceElement.toString()`, in order to include the classloader name and module version.

Or, I can include some commenting that explains why they are initialized to true.

Reply:
Am I missing something? Setting them to false in this constructor doesn't change the behavior since moduleName and classLoaderName are both null.

Reply:
This changes the behaviour because if the `classLoaderName` and `moduleVersion` aren't null/empty, then that information won't get printed since the values of `includeClassLoaderName` and `includeModuleVersion` will be false.

`setIncludeInfoFlags()` below is only used by `StackFrame`, since its stack trace output differs from other public constructor StackTraceElement objects.

Reply:
Public constructor `StackTraceElement` objects will always include the `classloaderName`/`moduleVersion` unless they are null/empty, with the exception of `StackFrame`, which uses the public constructor for `StackTraceElement`, but follows the same stack trace info rules as natively constructed `Throwable`.

Reply:
Since there aren't any methods to set the values of moduleVersion, classLoaderName, and this constructor sets them to null, basically includeModuleVersion and includeClassLoaderName are set to true in case the values of moduleVersion or classLoaderName are set via reflection?

Reply:
Note setting them to true is fine, I'm not arguing to change that, I was just making sure I understood.

Reply:
> Since there aren't any methods to set the values of moduleVersion, classLoaderName, and this constructor sets them to null, basically includeModuleVersion and includeClassLoaderName are set to true in case the values of moduleVersion or classLoaderName are set via reflection?

Yes, the flags are set to true here to cover if `classLoaderName` and `moduleVersion` end up not being null at the time of printing